libsawtooth/src/client/rest.rs

```diff
@@ -149,6 +152,124 @@ impl SawtoothClient for RestApiSawtoothClient {
 
         Ok(get::<StatusList>(&url, error_msg)?.map(convert_status_list))?.transpose()
     }
+    /// Send one or more batches to the sawtooth REST API to be submitted to the validator
+    fn submit_batches(
+        &self,
+        filename: String,
+        wait: Option<Duration>,
+        size_limit: usize,
+    ) -> Result<Vec<String>, SawtoothClientError> {
+        let mut url = format!("{}/batches", &self.url);
+        if let Some(wait_time) = wait {
+            url = url + &format!("?wait={}", wait_time.as_secs().to_string());
+        }
+
+        let mut batch_file = File::open(filename).map_err(|err| {
+            SawtoothClientError::new_with_source("failed to open batch file", err.into())
+        })?;
+
+        let batch_list: BatchList =
+            protobuf::parse_from_reader(&mut batch_file).map_err(|err| {
+                SawtoothClientError::new_with_source("unable to parse file contents", err.into())
+            })?;
+        let len = batch_list.batches.len();
+        let batch_ids = batch_list
+            .batches
+            .iter()
+            .map(|batch| batch.header_signature.clone())
+            .collect();
+
+        let batch_lists = split_batches(batch_list, size_limit);
+
+        let start = Instant::now();
+
+        for list in batch_lists.into_iter() {
+            let submit_list = list.write_to_bytes().map_err(|err| {
+                SawtoothClientError::new_with_source(
+                    "failed to get bytes from batch list",
+                    err.into(),
+                )
+            })?;
+            let request = Client::new()
+                .post(&url)
+                .header(header::CONTENT_TYPE, "application/octet-stream")
+                .body(submit_list);
+            let response = request.send().map_err(|err| {
+                SawtoothClientError::new_with_source("request failed", err.into())
+            })?;
+            if response.status().as_u16() != 202 {
+                let status = response.status();
+                let msg: ErrorResponse = response.json().map_err(|err| {
+                    SawtoothClientError::new_with_source(
+                        "failed to deserialize error response body",
+                        err.into(),
+                    )
+                })?;
+                return Err(SawtoothClientError::new(&format!("{} {}", status, msg)));
+            }
+        }
+        info!(
+            "batches: {},  batch/sec: {}",
+            len,
+            len as f64 / (((start.elapsed()).as_millis() as f64) / 1000f64)
+        );
+        Ok(batch_ids)
+    }
+}
+
+/// Check the status of the batches with the given ids, wait the requested amount of time
+/// for the batches to commit
+pub fn wait_for_batches(
```

Comment:
Looks like this isn't used anymore; should probably be removed.

Reply:
Fixed